pkg/proxy/iptables/proxier.go

```diff
@@ -643,6 +644,17 @@ func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
 		return
 	}
 
+	// kube-proxy in LocalModeNodeCIDR mode may cache stale Node.PodCIDR.
+	// It should restart on those situations to reconcile the state.
+	// Ref: https://issues.k8s.io/111321
```

Comment:
Do we know for sure that a kp restart on UID does not have other unintended consequences? I do like the big hammer though if we have a good sense that this won't cause problems when backporting.

Reply:
Restarting kube-proxy at any time ought to be totally safe. It will pause updates for however long it takes to restart and resync the caches, but other than that it should be a no-op.

Reply:
+1